### PR TITLE
3625 downtime msg

### PR DESCRIPTION
**Pull request by allthesignals**
Now:
Screen Shot 2021-06-28 at 12 20 03 PM

When in maintenance mode:
c56031dc-34c5-45f9-adcf-7212faaeaa48

#### Tasks/Bug Numbers
 - Fixes [AB#3625](https://dev.azure.com/NYCPlanning/cc280b0d-40a0-4689-b852-2e6247f1af50/_workitems/edit/3625)

### CODE REVIEW

File: client/app/templates/components/auth/login-dialog.hbs [hunk 1]
```diff
@@ -14,9 +14,14 @@
...
 <p>Once you're signed in, you'll be redirected back here, where you'll see your list of assigned projects.</p>
 
+{{#if this.hasUpcomingMaintenance}}
+  <h1>Please note: This portal will be down for 24 hours of system maintenance starting Monday, June 28 at 7PM.</h1>
+{{/if}}
+
 <a
-  class="button large expanded"
-  href={{this.oauthEndpoint}}
+  class="button large expanded {{if this.isMaintenancePeriod 'disabled'}}"
```
Comment (godfreyyeung) [inline client/app/templates/components/auth/login-dialog.hbs:22]: Can we just hide the button instead?

File: client/app/components/auth/login-dialog.js [hunk 2]
```diff
@@ -1,6 +1,22 @@
 import Component from '@ember/component';
+import { computed } from '@ember/object';
 import ENV from 'labs-zap-search/config/environment';
 
+const MAINTENANCE_RANGE = ['06/28/2021 19:00', '06/29/2021 19:00'];
```
Comment (godfreyyeung) [inline client/app/components/auth/login-dialog.js:5]: What timezone is this? Should you use UTC timestamps instead? Does this default to UTC?

**Review by godfreyyeung — APPROVED** [review]

File: client/app/components/auth/login-dialog.js (hunk 2 above)
Comment (allthesignals) [reply client/app/components/auth/login-dialog.js:5]: probably. browsers defaults to the user's local timezone

File: client/app/templates/components/auth/login-dialog.hbs (hunk 1 above)
Comment (allthesignals) [reply client/app/templates/components/auth/login-dialog.hbs:22]: we _can_

**Pull request merged**